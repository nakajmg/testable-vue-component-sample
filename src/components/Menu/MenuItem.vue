<template>
  <div class="MenuItem">
    <span class="MenuItem_Label"
      @click="clickMenuItem"
    >
      {{label}}
    </span>
  </div>
</template>

<script>
export default {
  name: "MenuItem",
  props: {
    label: String,
    name: String,
  },
  methods: {
    clickMenuItem() {
      this.$emit("clickMenuItem", { name: this.name })
    },
  },
}
</script>

<style lang="scss">
.MenuItem {
  &_Label {
    cursor: pointer;
    &:hover {
      text-decoration: underline;
    }
  }
}
</style>
