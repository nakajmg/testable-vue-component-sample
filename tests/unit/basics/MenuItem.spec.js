import { shallowMount } from "@vue/test-utils"
import MenuItem from "@/basics/MenuItem.vue"
import menuItems from "../../_mockData/menuItems.json"
describe("MenuItem.vue", () => {
  const propsData = menuItems[0]
  it("props", () => {
    const wrapper = shallowMount(MenuItem, { propsData })
    expect(wrapper.vm.$props).toEqual(propsData)
  })
  describe("methods", () => {
    it("clickMenuItem", () => {
      const wrapper = shallowMount(MenuItem, { propsData })
      wrapper.vm.clickMenuItem()
      expect(wrapper.emitted("clickMenuItem")).toBeTruthy()
      expect(wrapper.emitted("clickMenuItem")[0][0]).toEqual({
        name: propsData.name,
      })
    })
  })
  describe("template", () => {
    it("@click=clickMenuItem", () => {
      const mock = jest.fn()
      const wrapper = shallowMount(MenuItem, { propsData })
      wrapper.setMethods({
        clickMenuItem: mock,
      })
      wrapper.find(".MenuItem_Label").trigger("click")
      expect(mock).toHaveBeenCalled()
    })
    it("snapshot", () => {
      const wrapper = shallowMount(MenuItem, { propsData })
      expect(wrapper.vm.$el).toMatchSnapshot()
    })
  })
})
