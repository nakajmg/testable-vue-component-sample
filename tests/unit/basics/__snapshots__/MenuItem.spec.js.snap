// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem.vue template snapshot 1`] = `
<div
  class="MenuItem"
>
  <span
    class="MenuItem_Label"
  >
    
    About
  
  </span>
</div>
`;
